apps/eew/scfinder/main.cpp

```diff
@@ -592,56 +622,29 @@ class App : public Client::StreamApplication {
 											Core::Time(pga.get_timestamp()).iso().c_str(),
 											pga.get_value());
 						}
-						SEISCOMP_DEBUG("epicentral PGA value: %f cm/s/s",epicentralPGAvalue);	
+						SEISCOMP_DEBUG("Epicentral PGA value: %f cm/s/s",epicentralPGAvalue);	
+						SEISCOMP_DEBUG("Total number of PGA: %lu ",originBasedLatestMaxPGAs.size());	
 
 						// get the current time for this new finder object
 						long event_id = (long)time(NULL);
 						
-						// make constructor take a Coordinate designated by *cit...it will increment Nfinder
-						_finderList.push_back(new Finder(Coordinate(org->latitude().value(), 
-						                                            org->longitude().value()), 
-						                                 _latestMaxPGAs, 
-														 event_id, 
-														 _bufferLength.seconds()));
-
-						
-						// Use epicentral pga 
-						_processingOrigin = true;		
+						// make constructor from Coordinate
+						Finder* originBasedFinder = new Finder(Coordinate(org->latitude().value(), 
+						                                                  org->longitude().value()), 
+																originBasedLatestMaxPGAs, 
+																event_id, 
+																_bufferLength.seconds());							
 
 						// Call Finder
-						Core::Time tick = Core::Time::GMT();
-						_finderList.back()->process(tick, _latestMaxPGAs);			
-
-						// Skip epicentral pga 			
-						_processingOrigin = false;	
+						Core::Time tick = Core::Time::GMT();	
+						SEISCOMP_DEBUG("originBasedFinder->process...");						
+						originBasedFinder->process(tick, originBasedLatestMaxPGAs);		
+						SEISCOMP_DEBUG("originBasedFinder->process is Done");
 
 						// Send Finder solution
-						sendFinder(_finderList.back());
-						
-						// Remove external origin
-						_finderList.pop_back();
-
-						// Remove epicentral PGAs (neppgas last PGAs)
-						for ( int i = 0; i < neppgas ; ++i ) {
-							_latestMaxPGAs.pop_back(); 
-						}
-
-						// Check epicentral PGAs removed
-						for ( size_t i = 0; i < _latestMaxPGAs.size(); ++i ) {
-							PGA_Data &pga = _latestMaxPGAs[i];
-							if ( strcmp( pga.get_network().c_str(), "EP" ) != 0 ) {
-								continue;
-							}
-							SEISCOMP_DEBUG("WARNING: epicentral PGA still in buffer for %s.%s.%s.%s %f %f %s %f",
-											pga.get_network().c_str(),
-											pga.get_name().c_str(),
-											pga.get_location_code().c_str(),
-											pga.get_channel().c_str(),
-											pga.get_location().get_lat(),
-											pga.get_location().get_lon(),
-											Core::Time(pga.get_timestamp()).iso().c_str(),
-											pga.get_value());
-						}
+						SEISCOMP_DEBUG("sendFinder(originBasedFinder)...");	
+						sendFinder(originBasedFinder);
+						SEISCOMP_DEBUG("sendFinder(originBasedFinder) is Done");
```

Comment:
@jenrandrews @ sceylan : is it correct that originBasedFinder and originBasedLatestMaxPGAs will be destroyed once this method return ?

Reply:
I took a look at the full file and I think originBasedLatestMaxPGAs is truly locally scoped and will be destroyed. For originBasedFinder you are creating it with a "new" which means you should also call "delete" when done. So just add the line "delete originBasedFinder;" whenever in the function/code you are finished with the object (looks like line 647). The object looks locally scoped though, so you could also just create it with "Finder originBasedFinder = Finder(...);" and switch to "originBasedFinder.process()" and "sendFinder(&originBasedFinder)". Either approach should work =)